ocaml/xapi/repository.ml

```diff
@@ -695,11 +702,22 @@ let apply_updates' ~__context ~host ~updates_info ~livepatches ~acc_rpm_updates
   in
   (* Update states in cache *)
   update_cache ~host ~failed_livepatches ;
-  List.iter
-    (fun g -> debug "mandatory pending_guidance: %s" (to_string g))
-    mandatory ;
-  let ops = get_ops_of_pending ~__context ~host ~kind:Mandatory in
-  set_pending_guidances ~ops ~coming:mandatory ;
+  (* Set pending guidance lists *)
+  let set_guidance ~kind ~coming =
+    List.iter
+      (fun g ->
+        debug "setting %s pending guidance: %s" (kind_to_string kind)
+          (to_string g)
+      )
+      coming ;
+    let ops = get_ops_of_pending ~__context ~host ~kind in
+    set_pending_guidances ~ops ~coming
+  in
+  guidance |> List.iter (fun (kind, coming) -> set_guidance ~kind ~coming) ;
+  let get_livepatch_component (lp, _) = lp.LivePatch.component in
+  let applied = List.map get_livepatch_component applied_livepatches in
+  let failed = List.map get_livepatch_component failed_livepatches in
+  update_livepatch_failures ~__context ~host ~applied ~failed ;
```

Comment:
`update_livepatch_failures` -> `update_livepatch_failure_guidances`
I think it would be better to make name easy to understand.